### PR TITLE
01789 d hip17 unhappy tokens

### LINKED ISSUE #1789: Create test matrix for HIP-17 operations vs. unhappy token states
:bangbang: Should be broken down into sub-issues for easier tracking.

### Problem

HIP-17 affects all HTS operations (including three new queries), as well as `CryptoTransfer`.

In all cases these operations involve tokens; and besides the "happy" state of a non-expired, non-deleted token, a token may be:
- [ ] Expired.
- [ ] Deleted.
- [ ] Auto-removed.

We need to validate (and characterize via a regression test) the behavior of each operation when it faces a token in a non-happy state.

### Solution

Create a devoted `Hip17UnhappyTokensSuite` with a spec characterizing how each HIP-17 operation responds to each case.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by ljianghedera**
**Description**:
This PR added the EET test scenarios for NFT token when the token is deleted.

**Related issue(s)**:

Fixes #1789

**Notes for reviewer**:
Right now,  token auto removal and expiry features are still not implemented completely, the test cases for those scenarios will be added when they are finished.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by tinker-michaelj — APPROVED** [review]
Looks perfect! 🥂

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+								.adminKey(ADMIN_KEY)
+								.treasury(TOKEN_TREASURY),
+						tokenAssociate(ANOTHER_USER, NFTdeleted)
+				)
+				.when(
+						tokenDelete(NFTdeleted)
+				)
+				.then(
+						tokenDissociate(ANOTHER_USER, NFTdeleted)
+								.hasKnownStatus(SUCCESS)
```
Comment (qnswirlds) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java:273]: We should expect `TOKEN_WAS_DELETED` right?

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+								.supplyKey(SUPPLY_KEY)
+								.adminKey(ADMIN_KEY)
+								.treasury(TOKEN_TREASURY),
+						tokenAssociate(ANOTHER_USER, NFTdeleted)
+				)
+				.when(
+						tokenDelete(NFTdeleted)
+				)
+				.then(
+						tokenDissociate(ANOTHER_USER, NFTdeleted)
```
Comment (abhishek-hedera) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java:272]: Shouldn't this fail with `TOKEN_WAS_DELETED`?

File: test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java (hunk 1 above)
Comment (ljianghedera) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java:273]: Current implementation gives a SUCCESS for this operation. I will look into more and decide whether it's an issue or not.

Comment (ljianghedera) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/token/Hip17UnhappyTokensSuite.java:273]: @qnswirlds seems like this is right behavior, do you think we need more research here?

**Review by qnswirlds — APPROVED** [review]

**Pull request merged**